docs/revsets.md

```diff
@@ -135,6 +135,13 @@ revsets (expressions) as arguments.
 * `present(x)`: Same as `x`, but evaluated to `none()` if any of the commits
   in `x` doesn't exist (e.g. is an unknown branch name.)
 
+## String patterns
+
+Functions that perform string matching support the following pattern syntax.
+
+* `"needle"`: The haystack contains `needle`.
```

Comment:
Ok I know the saying, and it even directly translates to the same saying in my native language, but still I feel like technical "The string contains a substring `not needle`" would be sliiightly less confusing to the reader, especially as there is zero explicit association between "haystack" (suppose I don't know what that word means) and the string we're trying to match.

Reply:
That doesn't make sense to me...
How about "container to search" and "what to search for"?

Reply:
Maybe:
* `"substring"`: Matches strings that contain `substring`.
* `literal:"string"`:  Matches strings exactly equal to `string`.

There's only one other public-facing use of `needle` in the docs as far as I can tell (in [docs/templates.md](https://github.com/martinvonz/jj/blob/313670d3c22ee4ddb715cb4e2d736bedcaa4368f/docs/templates.md?plain=1#L140)), so I don't see a strong motivation to keep using the word.

Also, many of the uses of `(something_)pattern` above still use the verb "contains" or "is" when "matches" and "matching" would be more appropriate with the new pattern syntax.

Reply:
> * `"substring"`: Matches strings that contain `substring`.
> * `literal:"string"`:  Matches strings exactly equal to `string`.

Much better! Thanks for the suggestion.